### PR TITLE
Migrate glib gstreamer, update OSX  compiler for 5.12

### PR DESCRIPTION
**Pull request by hmaarrfk**
Closes #174 #176
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
Seems to depend on https://github.com/conda-forge/guile-feedstock/pull/6

**Comment by hmaarrfk** [comment]
Now with glib unpinned, i get
```
The reported errors are:
⇟   Encountered problems while solving.
   Problem: cannot install both binutils_impl_linux-64-2.35.1-h193b22a_1 and binutils_impl_linux-64-2.35.1-h17ad2fc_1
   Problem: cannot install both ld_impl_linux-64-2.35.1-hed1e6ac_1 and ld_impl_linux-64-2.35.1-hea4e1c9_1

```

any ideas @isuruf

**Comment by hmaarrfk** [comment]
Probably needs: https://github.com/conda-forge/glib-feedstock/pull/80

**Comment by hmaarrfk** [comment]
@isuruf would you be OK merging this, or do you prefer we do this, 2.64.6->2 migration properly?

**Comment by isuruf** [comment]
I'd rather we fix glib issue properly.

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
OSX
```
2020-12-18T14:33:44.0246050Z > clang-11: error: invalid linker name in argument '-fuse-ld=gold'
2020-12-18T14:33:44.0246780Z test config.qtbase.tests.use_gold_linker FAILED
```

**Comment by hmaarrfk** [comment]
@conda-forge/qt this is mostly ready for linux.

This migration is necessary to move glib forward, a linux only dependency.

I personally don't think it is important that we rebuild for windows or OSX.

OSX seems to have compiler issues.

File: recipe/build.sh [hunk 1]
```diff
@@ -294,6 +294,7 @@ if [[ ${HOST} =~ .*darwin.* ]]; then
                 -no-egl \
                 -no-openssl \
                 -optimize-size \
+                -no-use-gold-linker \
```
Comment (isuruf) [inline recipe/build.sh:297]: Gold linker thing is not the error

Comment (hmaarrfk) [reply recipe/build.sh:297]: sorry, brain fart.

Should we try to add `-Welaborated-enum-base` or `-Wno-elaborated-enum-base`

**Comment by hmaarrfk** [comment]
Maybe clues can be understood from this comment
https://github.com/darlinghq/darling/issues/856#issuecomment-667765874

**Comment by hmaarrfk** [comment]
Hooray! 🥳

**Comment by hmaarrfk** [comment]
Were you going to manually build linux and OSX?

**Comment by isuruf** [comment]
I'll do the linux build. Anybody wants to do the OSX build?
cc @conda-forge/core

**Comment by chrisburr** [comment]
> Anybody wants to do the OSX build?

I've just started the build on the mac mini.

**Comment by chrisburr** [comment]
Done [osx_64_.zst.log](https://github.com/conda-forge/qt-feedstock/files/5732255/osx_64_.zst.log)

**Pull request merged**